src/components/ProfilePage/QuestsOverview/QuestsOverview.tsx

```diff
@@ -0,0 +1,94 @@
+import { useTranslation } from 'react-i18next';
+import { useOngoingQuests } from 'src/hooks/useOngoingQuests';
+import type { Quest } from 'src/types/loyaltyPass';
+import { checkInclusion } from '../checkInclusion';
+import { QuestCard } from '../QuestCard/QuestCard';
+import {
+  QuestsOverviewContainer,
+  QuestsOverviewTitle,
+} from './QuestsOverview.style';
+
+interface QuestCarouselProps {
+  pastCampaigns?: string[];
+  traits?: string[];
+  // traits?: Trait[];
+}
+
+export const QuestsOverview = ({
+  pastCampaigns,
+  traits,
+}: QuestCarouselProps) => {
+  const { t } = useTranslation();
+  const { quests, isLoading: isQuestsLoading, url } = useOngoingQuests();
+
+  return (
+    <QuestsOverviewContainer>
+      <QuestsOverviewTitle variant="headerMedium">
+        {t('missions.available')}
+      </QuestsOverviewTitle>
+      {/** render quests */}
+      {!isQuestsLoading
+        ? quests?.map((quest: Quest, index: number) => {
+            const baseURL = quest?.attributes?.Image?.data?.attributes?.url;
+            const imgURL = baseURL && new URL(baseURL, url.origin);
+            const rewards = quest?.attributes?.CustomInformation?.['rewards'];
+            const rewardType =
+              quest?.attributes?.CustomInformation?.['rewardType'];
+            const rewardRange =
+              quest?.attributes?.CustomInformation?.['rewardRange'];
+            const chains = quest?.attributes?.CustomInformation?.['chains'];
+            const claimingIds =
+              quest?.attributes?.CustomInformation?.['claimingIds'];
+            const rewardsIds =
+              quest?.attributes?.CustomInformation?.['rewardsIds'];
+            const questTraits =
+              quest?.attributes?.CustomInformation?.['traits'];
+            const endDate = quest?.attributes?.EndDate;
+            const startDate = quest?.attributes?.StartDate;
+            //todo: exclude in a dedicated helper function
+            let completed = false;
+            if (rewardsIds && pastCampaigns) {
+              completed = checkInclusion(pastCampaigns, rewardsIds);
+            }
+
+            let isUnlockedForUser = false;
+            if (questTraits?.length > 0 && traits) {
+              isUnlockedForUser = checkInclusion(traits, questTraits);
+              // isUnlockedForUser = false;
+            }
+
+            return (
+              <QuestCard
+                key={`available-mission-${index}`}
+                title={quest?.attributes?.Title}
```

Comment:
quest should be passed instead of each of its attributes, will make things muh easier to re-use and will increase the types thoughness

Reply:
I like that idea and you taught me that before: Single Responsibility Principle -->

But here I am not sure if we can make it work so easily because the input differs depending on what kind of QuestCard is shown. That´s why I preferred to pass in the props, rather than handling them inside the QuestCard component itself. Wdyt?

<img width="852" alt="Screenshot 2024-12-19 at 10 36 28" src="https://github.com/user-attachments/assets/ed1b9839-5b5b-4e96-8b7c-0b547124733a" />

<img width="438" alt="Screenshot 2024-12-19 at 10 34 55" src="https://github.com/user-attachments/assets/12276286-76a2-4b76-8036-b3b6ff7ec2dd" />

<img width="705" alt="Screenshot 2024-12-19 at 10 34 19" src="https://github.com/user-attachments/assets/76eff405-8c3a-467a-a6be-05a8797481f6" />

Reply:
I do not see the issue sorry, just pass different data to it following the same interface

Reply:
So, you are suggesting to always pass in an object into the QuestCard which shares the same structure. It´s done now @tcheee @oktapodia